packages/babel-types/src/definitions/core.js

```diff
@@ -836,6 +836,10 @@ defineType("VariableDeclarator", {
     id: {
       validate: assertNodeType("LVal"),
     },
+    exclamation: {
```

Comment:
Maybe we should call it `definiteAssertion` (or `assertDefined`) instead of `exclamation`? I saw that tsc uses `node.exclamationToken`, but it also uses `node.questionToken` instead of `node.optional`.

Reply:
In theory, I'd stick to the syntactic characteristics of a token to ease later semantic changes to its meaning, i.e. name it `exclamation[Token]` instead of `definite` since token definitions may later be broadened to include more than the original intentions—and the parser should add as little semantic overlay/interpretation as possible, IMHO.

However, since there is precedent for using semantic descriptions even when the TS AST uses a syntactic naming scheme (`questionToken` → `optional`), following that same path again may be the better way with regard to internal consistency.

In any case, my feelings are not very strong either way, so I'd be fine with both.